pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -295,3 +327,37 @@ func renameTeam(ctx context.Context, workflowCtx *workflow.Context, atlasTeam *m
 
 	return atlasTeam, nil
 }
+
+func teamsManagedByAtlas(ctx context.Context, atlasClient mongodbatlas.Client, orgID string) customresource.AtlasChecker {
+	return func(resource v1.AtlasCustomResource) (bool, error) {
+		team, ok := resource.(*v1.AtlasTeam)
+		if !ok {
+			return false, errors.New("failed to match resource type as AtlasTeams")
+		}
+
+		if team.Status.ID == "" {
+			return false, nil
+		}
+
+		atlasTeam, _, err := atlasClient.Teams.Get(ctx, orgID, team.Status.ID)
+		if err != nil {
+			var apiError *mongodbatlas.ErrorResponse
+			if errors.As(err, &apiError) && (apiError.ErrorCode == atlas.NotInGroup || apiError.ErrorCode == atlas.ResourceNotFound) {
```

Comment:
Are we sure these are the appropriate error codes? Just cautious because DF used different ones...

Reply:
Yes, I do. These are the possible ones for 404 errors.